File: support_tools/swigwin-3.0.12/Examples/test-suite/csharp/director_protected_runme.cs
using System;

namespace director_protectedNamespace {

public class runme
{
  static void Main() 
  {
    runme r = new runme();
    r.run();
  }

  void run()
  {
    Bar b  = new Bar();
    Foo f = b.create();
    FooBar fb = new FooBar();
    FooBar2 fb2 = new FooBar2();
    FooBar3 fb3 = new FooBar3();

    String s;
    s = fb.used();
    if ( s != ("Foo::pang();Bar::pong();Foo::pong();FooBar::ping();"))
      throw new Exception("bad FooBar::used" + " - " + s);

    s = fb2.used();
    if ( s != ("FooBar2::pang();Bar::pong();Foo::pong();FooBar2::ping();"))
      throw new Exception("bad FooBar2::used");

    s = b.pong();
    if ( s != ("Bar::pong();Foo::pong();Bar::ping();"))
      throw new Exception("bad Bar::pong");

    s = f.pong();
    if ( s != ("Bar::pong();Foo::pong();Bar::ping();"))
      throw new Exception("bad Foo::pong");

    s = fb.pong();
    if ( s != ("Bar::pong();Foo::pong();FooBar::ping();"))
      throw new Exception("bad FooBar::pong");

//    if (fb3.cheer() != "FooBar3::cheer();")
//      throw new Exception("bad fb3::cheer");

    if (fb2.callping() != "FooBar2::ping();")
      throw new Exception("bad fb2.callping");

    if (fb2.callcheer() != "FooBar2::pang();Bar::pong();Foo::pong();FooBar2::ping();")
      throw new Exception("bad fb2.callcheer");

    if (fb3.callping() != "Bar::ping();")
      throw new Exception("bad fb3.callping");

    if (fb3.callcheer() != "FooBar3::cheer();")
      throw new Exception("bad fb3.callcheer");
  }
}

class FooBar : Bar
{
  public FooBar() : base()
  {
  }

  protected override String ping()
  {
    return "FooBar::ping();";
  }
}

class FooBar2 : Bar
{
  public FooBar2() : base()
  {
  }

  protected override String ping()
  {
    return "FooBar2::ping();";
  }

  protected override String pang()
  {
    return "FooBar2::pang();";
  }
}

class FooBar3 : Bar
{
  public FooBar3() : base()
  {
  }

  protected override String cheer()
  {
    return "FooBar3::cheer();";
  }
}

}

File: support_tools/swigwin-3.0.12/Examples/test-suite/php/director_protected_runme.php
<?php

require "tests.php";
require "director_protected.php";

check::functions(array(foo_pong,foo_s,foo_q,foo_ping,foo_pang,foo_used,foo_cheer,bar_create,bar_callping,bar_callcheer,bar_cheer,bar_pong,bar_used,bar_ping,bar_pang,a_draw,b_draw));
check::classes(array(Foo,Bar,PrivateFoo,A,B,AA,BB));
check::globals(array(bar_a));

class FooBar extends Bar {
  protected function ping() {
    return "FooBar::ping();";
  }
}

class FooBar2 extends Bar {
  function ping() {
    return "FooBar2::ping();";
  }

  function pang() {
    return "FooBar2::pang();";
  }
}

class FooBar3 extends Bar {
  function cheer() {
    return "FooBar3::cheer();";
  }
}

$b = new Bar();
$f = $b->create();
$fb = new FooBar();
$fb2 = new FooBar2();
$fb3 = new FooBar3();

check::equal($fb->used(), "Foo::pang();Bar::pong();Foo::pong();FooBar::ping();", "bad FooBar::used");

check::equal($fb2->used(), "FooBar2::pang();Bar::pong();Foo::pong();FooBar2::ping();", "bad FooBar2::used");

check::equal($b->pong(), "Bar::pong();Foo::pong();Bar::ping();", "bad Bar::pong");

check::equal($f->pong(), "Bar::pong();Foo::pong();Bar::ping();", "bad Foo::pong");

check::equal($fb->pong(), "Bar::pong();Foo::pong();FooBar::ping();", "bad FooBar::pong");

$method = new ReflectionMethod('Bar', 'ping');
check::equal($method->isProtected(), true, "Foo::ping should be protected");

$method = new ReflectionMethod('Foo', 'ping');
check::equal($method->isProtected(), true, "Foo::ping should be protected");

$method = new ReflectionMethod('FooBar', 'pang');
check::equal($method->isProtected(), true, "FooBar::pang should be protected");

$method = new ReflectionMethod('Bar', 'cheer');
check::equal($method->isProtected(), true, "Bar::cheer should be protected");

$method = new ReflectionMethod('Foo', 'cheer');
check::equal($method->isProtected(), true, "Foo::cheer should be protected");

check::equal($fb3->cheer(), "FooBar3::cheer();", "bad fb3::pong");
check::equal($fb2->callping(), "FooBar2::ping();", "bad fb2::callping");
check::equal($fb2->callcheer(), "FooBar2::pang();Bar::pong();Foo::pong();FooBar2::ping();", "bad fb2::callcheer");
check::equal($fb3->callping(), "Bar::ping();", "bad fb3::callping");
check::equal($fb3->callcheer(), "FooBar3::cheer();", "bad fb3::callcheer");

check::done();
?>
